clang/lib/CodeGen/CGOpenMPRuntime.cpp

```diff
@@ -2331,8 +2332,10 @@ void CGOpenMPRuntime::emitBarrierCall(CodeGenFunction &CGF, SourceLocation Loc,
   auto *OMPRegionInfo =
       dyn_cast_or_null<CGOpenMPRegionInfo>(CGF.CapturedStmtInfo);
   if (CGF.CGM.getLangOpts().OpenMPIRBuilder) {
-    CGF.Builder.restoreIP(OMPBuilder.createBarrier(
-        CGF.Builder, Kind, ForceSimpleCall, EmitChecks));
+    auto Result = OMPBuilder.createBarrier(CGF.Builder, Kind, ForceSimpleCall,
```

Comment:
_nit#1_: I have been receiving feedback on some of my PRs to be a little more conservative with the use of `auto`. I have concluded that it is best to use it only when it is obvious from the RHS of the assignment (e.g `dyn_cast<SomeType>`. [(Relevant part of the style guide)](https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable)

Reply:
The reason I've tended towards `auto` in this patch is that the full type in this case is `llvm::Expected<llvm::OpenMPIRBuilder::InsertPointTy>`, which is quite verbose, so I thought it was justified in this case for readability. But you're right, it doesn't look like the style guide is making an exception for such cases, so I'll look into updating the PR based on this comment.

Maybe adding an `InsertPointOrErrorTy` typedef would be a good way to address this.

Reply:
This is done now, thanks for the suggestion.